onnxruntime/core/providers/cuda/math/binary_elementwise_ops.cc

```diff
@@ -465,6 +465,69 @@ Status Pow::ComputeInternal(OpKernelContext* context) const {
   return s;
 }
 
+ONNX_OPERATOR_VERSIONED_KERNEL_EX(
+    Mod, kOnnxDomain, 10, 12, kCudaExecutionProvider,
+    (*KernelDefBuilder::Create())
+        .TypeConstraint("T",
+                        BuildKernelDefConstraints<int32_t, int64_t, uint32_t, uint64_t, float, double, MLFloat16>()),
+    Mod);
+
+ONNX_OPERATOR_KERNEL_EX(Mod, kOnnxDomain, 13, kCudaExecutionProvider,
+                        (*KernelDefBuilder::Create())
+                            .TypeConstraint("T", BuildKernelDefConstraints<int32_t, int64_t, uint32_t, uint64_t, float,
+                                                                           double, MLFloat16, BFloat16>()),
+                        Mod);
+
+Status Mod::ComputeInternal(OpKernelContext* context) const {
+  namespace on = ONNX_NAMESPACE;
+  BinaryElementwisePreparation prepare;
+  ORT_RETURN_IF_ERROR(Prepare(context, &prepare));
+  auto element_type = prepare.lhs_tensor->GetElementType();
+  ORT_ENFORCE(fmod_ || element_type == on::TensorProto_DataType_INT32 ||
+                  element_type == on::TensorProto_DataType_INT64 || element_type == on::TensorProto_DataType_UINT32 ||
+                  element_type == on::TensorProto_DataType_UINT64,
+              "Non-fmod can support integer types only.");
+#define CASE_MOD_ELEMENT_TYPE(name, onnx_type, data_type)                                                           \
```

Comment:
I am guessing the intent behind this macro is so that we can call `Impl_##name`. Is it possible to avoid using such macros?

Reply:
The macro is to avoid duplicated code. If not using it, the file will be too long with lots of duplicated codes. And next time if you want to change any of the arguments, you need to change all the places, which is not easy to maintain.

Reply:
Are we able to use lambdas by passing in std::function if needed?

Reply:
Lambda will not work for class template parameters... Otherwise we can make it a function instead of Macro.

Reply:
yeah, I feel having a function might be cleaner. We can get that in another PR. Thanks.